src/mono/mono/mini/aot-compiler.c

```diff
@@ -15122,17 +15113,13 @@ aot_assembly (MonoAssembly *ass, guint32 jit_opts, MonoAotOptions *aot_options)
 		acfg->flags = (MonoAotFileFlags)(acfg->flags | MONO_AOT_FILE_FLAG_LLVM_ONLY);
 
 	acfg->assembly_name_sym = g_strdup (get_assembly_prefix (acfg->image));
-#ifdef TARGET_WASM
-	acfg->global_prefix = g_strdup_printf ("mono_aot_%s", g_strdup(mono_fixup_symbol_name (acfg->assembly_name_sym)));
-#else
 	char *p;
-    /* Get rid of characters which cannot occur in symbols */
+	/* Get rid of characters which cannot occur in symbols */
 	for (p = acfg->assembly_name_sym; *p; ++p) {
 		if (!(isalnum (*p) || *p == '_'))
 			*p = '_';
 	}
```

Comment:
shouldn't this use `mono_fixup_symbol_name` too?

Reply:
No, I initially I made here changes for `wasm` by mistake, so reverting in this PR.

Reply:
it's possible that we didn't hit the original issue here because this only deals with assembly names but I'd prefer using the same logic everywhere.